### DISCUSSION TITLE
APC protocol

### DISCUSSION THREAD

**Discussion by wheyVanilla** [syssi/esphome-apc-ups#19] [Q&A]
I have a recorded APC UPS stream txt file and I tried to utilize your code to decode it but can not find a entry point for the apc_ups.c. Is anywhere I can find a example to use it without ESPhome?

**Comment by syssi** [comment]
What do you try to achieve in detail?

**Reply by wheyVanilla** [reply]
I recorded a binary file from a APC UPS and I saw part of your code can decode the binary to string. I'm wondering how I can tailor this repo to decode my file.

**Reply by syssi** [reply]
And you don't want to use an ESP + ESPHome?

**Comment by wheyVanilla** [comment]
No, I don’t.

**Reply by syssi** [reply]
What kind of hardware do you want to use to interface the UPS? My implementation cannot he used a library. You have to extract/replicate the relevant parts.
